crates/relayer/relayer-components/src/relay/impls/event_relayers/packet_event.rs

```diff
@@ -65,6 +85,49 @@ where
             if MatchPacketDestinationChain::should_relay_packet(relay, &packet).await? {
                 relay.relay_packet(&packet).await?;
             }
+        } else if let Some(update_client_event) =
+            src_chain.try_extract_from_event(PhantomData::<SrcChain::UpdateClientEvent>, event)
+        {
+            let src_client_id = relay.src_client_id();
+            let client_state = src_chain
+                .query_client_state_with_latest_height(PhantomData, src_client_id)
+                .await
+                .map_err(Relay::raise_error)?;
+
+            match dst_chain
+                .check_misbehaviour(&update_client_event, &client_state)
+                .await
+            {
+                Ok(Some(evidence)) => {
+                    relay
+                        .log(
+                            "Found misbehaviour, will build message and submit",
+                            &LevelDebug,
+                        )
+                        .await;
+
+                    let msg = src_chain
+                        .build_misbehaviour_message(&evidence)
```

Comment:
regarding passing `Chain::ClientId`. why can't we pass it here 

```diff
- .build_misbehaviour_message(&evidence)
+ .build_misbehaviour_message(src_client_id, &evidence)
```

Reply:
Oh I missed that my bad